src/main/java/dan200/computercraft/core/filesystem/FileSystem.java

```diff
@@ -697,11 +729,7 @@ public void flush() throws IOException
                     throw new UnsupportedOperationException();
                 }
             };
-            synchronized( m_openFiles )
-            {
-                m_openFiles.add( file );
-            }
-            return file;
+            return (IMountedFileNormal) openFile( file, reader );
```

Comment:
```
private synchronized <T extends IMountedFile> T openFile( T file, Closeable handle ) throws FileSystemException
```
I think that would accurately capture what is trying to be done here.
It actually seems like a better solution.